src/upload/process.rs

```diff
@@ -395,6 +396,18 @@ pub async fn process_upload(args: UploadArgs) -> Result<()> {
         return Err(UploadError::Incomplete(message).into());
     }
 
+    // If hidden settings are enabled, update the hash value with the new cache file.
+    if let Some(ref hidden_settings) = config_data.hidden_settings {
```

Comment:
There is a problem in generating the hash during the `upload` command: if you are deploying a candy machine with collection, the cache file will be updated with the collection mint during `deploy` – this will make the hash of the cache file to change.